### DISCUSSION TITLE
Generating an FLP

### DISCUSSION THREAD

**Discussion by amd75692** [demberto/PyFLP#127] [Q&A] [answered]
Hi,

I've been using PyFLP for a few days. It's really amazing 😄

As far as i understand, PyFLP is meant to parse flp files. I was wondering if it's somehow possible to generate flp files like [jukeblocks.io](https://jukeblocks.io).

What features needs to be implemented in order to achieve this ?

Thank you.

**Comment by demberto** [comment, marked as answer]
*AFAICT* Jukeblocks doesn't generate a new FLPs, it takes a base FLP and adds / changes the necessary stuff in it.

An FLP cannot be generated from scratch unless you have 100% knowledge of the FLP format (which I don't, and people who do don't share it with the public).

If you are familiar with PyFLP's low-level [event API](https://pyflp.readthedocs.io/en/latest/reference/events.html), you can copy events of existing items, say a channel to add a new one. I am thinking of ways to make this easier, but the amount of unknowns I keep encountering in every situation is massive.

Thanks for the kind words :)

**Reply by amd75692** [reply]
I'm currently struggling with the low-level event API trying to append events in eventTree. I would really appreciate any piece of code achieving this copy 🙏.

**Reply by demberto** [reply]
I'll add something for it in the documentation soon.
